src/widgets/steps_widgets/extra_trees_vis.py

```diff
@@ -291,7 +288,7 @@ def make_graph(self, data: str):
     def click_listener(self, button_type: str):
```

Comment:
Maybe common declarations should be initialized in the parent class with NotImplementedError? Just to have a guarantee that writing the next algorithm any common function won't be omitted.